lucille-core/src/main/java/com/kmwllc/lucille/core/fileHandler/JsonFileHandler.java

```diff
@@ -112,10 +112,19 @@ public Document next() {
           } else {
             ObjectNode node = (ObjectNode) mapper.readTree(line);
             List<String> parts = new ArrayList<>(idFields.size());
+            List<String> blankIdFields = new ArrayList<>();
```

Comment:
We're doing this for every single document so we should try to be economical about creating new objects. In this case, I'd recommend not keeping a blankIdFields array but just logging a warning for each individual blank field as it is encountered.